http1.py

```diff
@@ -0,0 +1,76 @@
+import socket
+import sys
+
+
+class InvalidHTTPCodeError(Exception):
+    pass
+
+
+class NotGETRequestError(Exception):
+    pass
+
+
+class NotHTTP1_1Error(Exception):
+    pass
+
+
+class BadRequestError(Exception):
+    pass
+
+
+class HttpServer(object):
+    def http_server():
+        user_socket = socket.socket(
+            socket.AF_INET,
+            socket.SOCK_STREAM,
+            socket.IPPROTO_IP)
+        user_socket.bind(('127.0.0.1', 50000))
+        user_socket.listen(1)
+
+        while True:
+            conn, addr = user_socket.accept()
+            message = ''
+
+            while True:
+                msg_recieved = conn.recv(32)
+                message += msg_recieved
+                if len(msg_recieved) < 32:
+                    break
+            try:
+                # This seems to be a mac-specific bug
+                conn.shutdown(socket.SHUT_RD)
```

Comment:
This call informs the client that you will no longer be listening to anything they have to say.  It shouldn't be required here.  Can you describe the "mac-specific" bug you are seeing?